### PR TITLE
Allow setting StatsHandler as CallOption

### PR DESCRIPTION
**Pull request by dzbarsky**
@dfawley

### CODE REVIEW

**Comment by dzbarsky** [comment]
ok updated, i mostly did what you suggested but kept a unified StatsHandler in transport to avoid making more changes, what do you think?

File: rpc_util.go [hunk 1]
```diff
@@ -129,9 +129,15 @@ type callInfo struct {
 	maxReceiveMessageSize *int
 	maxSendMessageSize    *int
 	creds                 credentials.PerRPCCredentials
+	rpcStatsHandlers      []stats.RPCHandler
+	connStatsHandlers     []stats.ConnHandler
```
Comment (dfawley) [inline rpc_util.go]: We shouldn't need a slice of conn handlers.  Let's only do this if we must.

File: rpc_util.go [hunk 2]
```diff
@@ -129,9 +129,15 @@ type callInfo struct {
...
 }
 
-var defaultCallInfo = callInfo{failFast: true}
+var defaultCallInfo = callInfo{
+	failFast:          true,
+	rpcStatsHandlers:  make([]stats.RPCHandler, 0),
```
Comment (dfawley) [inline rpc_util.go]: Leaving these to default to "nil" is fine here.  Only maps need to be 'made'.

File: stats/handlers.go [hunk 3]
```diff
@@ -62,3 +65,8 @@ type Handler interface {
 	// HandleConn processes the Conn stats.
 	HandleConn(context.Context, ConnStats)
 }
+
+type Handler interface {
```
Comment (dfawley) [inline stats/handlers.go]: Travis is failing because there's no comment here.  You can keep the old comment.

**Comment by dzbarsky** [comment]
Will run gofmt before checking in, but I want this to compile/run tests first, not sure the transport changes are right

**Comment by dfawley** [comment]
Looks like there are still compilation problems.  Note that you can enable travis on your fork of grpc in your repo via https://travis-ci.org/profile/dzbarsky so you don't need to push updates to a PR to get travis to run.

**Comment by dfawley** [comment]
@dzbarsky, do you plan on picking this back up soon?  Thanks!

**Comment by dzbarsky** [comment]
Unfortunately I don't currently have time to work on this one, if you want to land it feel free to pick it up. Otherwise I can try to find someone else at Dropbox to finish it.

**Comment by dfawley** [comment]
We don't have the cycles to pick this up, and may not for some time.  Let's keep tracking using the issue (#1431).  Thanks!

**Pull request closed without merging**